routes/clients.rb

```diff
@@ -80,15 +80,25 @@ def self.registered(app)
                       contact_email: params[:contact_email] || client.contact_email,
                       contact_name: params[:contact_name] || client.contact_name,
                       active: params[:active] || client.active)
-        redirect "/clients/#{params[:id]}"
-      end
 
-      app.delete '/clients/:id/?' do
-        # delete a client
-        client = Client[id: params[:id]]
-        client.delete unless client.nil?
-        redirect '/clients' unless client.nil?
-        404
+        # if the active state is false, all projects are deleted and resources
+        # are re-assigned to the default project.
+        unless client.active || client.name == 'default'
+          unless client.projects.empty?
+            projects = client.projects
+            projects.each do |project|
+              resources = Report.get_resources(project)
+              unless resources.empty?
+                resources.each { |resource_type| resource_type.update(project_id: 1) }
```

Comment:
This is correct, but it would be safer to look up the default project's id by name instead of hard-coding 1 (I think in current production database, the id is actually 9)